### ISSUE TITLE
New Rule: SC2-4-7-focus-visible

### ISSUE THREAD

**Issue by john-urbilog** [act-rules/act-rules.github.io#339]
---
name: Focus visible
test_type: composite

description: |
 This rule checks that keyboard focus indicator is visible.
 
test_aspects:
- DOM Tree
- CSS Styling

success_criterion: 
- 2.4.7 # Focus Visible

atomic_rules: 
- TODO

authors:
- John Hicks
- Emma Pratt Richens
---

### Applicability

The rule applies to any HTML or SVG element that is [focusable] (#focusable).

**Note**: This rule only applies to HTML and SVG. Thus, it is a partial check for WCAG 2.0 success criterion 2.1.2, which applies to all content.

### Expectation

For each test target in focus this is indicate visibly.

**Note**: ...

## Assumptions

- TODO

## Accessibility support

There are no major accessibility support issues known for this rule.

## Background

- [Understanding Success Criterion 2.4.7: Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
- [F55: Failure of Success Criteria 2.1.1, 2.4.7, and 3.2.1 due to using script to remove focus when focus is received](https://www.w3.org/TR/WCAG20-TECHS/F55.html)
- [F78: Failure of Success Criterion 2.4.7 due to styling element outlines and borders in a way that removes or renders non-visible the visual focus indicator](https://www.w3.org/TR/WCAG20-TECHS/F78.html)

### Passed

```html
Code snippet to demonstrate any passed test cases.
```

### Failed

```html
Code snippet to demonstrate any failed test cases.
```

### Inapplicable

```html
Code snippet to demonstrate any inapplicable test cases.
```

**Comment by EmmaJP** [comment]
Thoughts so far on possible atomic tests...

- A test that checks the default browser styles for focus have not been removed or reset (F55).
- A test that checks there is a :focus style, even if the default is removed/replaced.
- A test that checks the element is visible when it has focus, even if it is hidden when not in focus.
- A test that checks the default browser styles are not covered or camouflaged (F78), which would have to be a manual test.

One of the first 2 would need to pass, as well as the others.

**Comment by EmmaJP** [comment]
---
name: Default browser focus style enabled
test_type: atomic

description: |
...

test_aspects:
- DOM Tree
- CSS Styling

authors:
- John Hicks
- Emma Pratt Richens
---
## Test procedure

### Applicability

Applicability of the rule

**Note**: Any Notes should follow this notation.

### Expectation

Expectations for the tests

## Assumptions

_There are currently no assumptions/ add any assumptions as necessary_

## Accessibility Support

_There are no major accessibility support issues known for this rule/ or add any support details as necessary_

## Background

- List of links to any additional reading resources

## Test Cases

### Passed

```html
Code snippet to demonstrate any passed test cases.
```

### Failed

```html
Code snippet to demonstrate any failed test cases.
```

### Inapplicable

```html
Code snippet to demonstrate any inapplicable test cases.
```

**Comment by jkodu** [comment]
PR for this rule is in here #415. Closing the issue.
Please feel free to re-open if necessary after reaching a consensus on the PR.
